pkg/cincinnati/cincinnati.go

```diff
@@ -146,17 +156,98 @@ func (c Client) GetUpdates(ctx context.Context, uri *url.URL, arch string, chann
 		}
 	}
 
-	var updates []Update
+	var updates []configv1.Release
 	for _, i := range nextIdxs {
-		updates = append(updates, Update(graph.Nodes[i]))
+		update, err := convertRetrievedUpdateToRelease(graph.Nodes[i])
+		if err != nil {
+			return current, nil, nil, &Error{
+				Reason:  "ResponseInvalid",
+				Message: fmt.Sprintf("invalid recommended update node: %s", err),
+			}
+		}
+		updates = append(updates, update)
+	}
+
+	var conditionalUpdates []configv1.ConditionalUpdate
+	for _, conditionalEdges := range graph.ConditionalEdges {
+		for _, edge := range conditionalEdges.Edges {
+			if version.String() == edge.From {
+				var target *node
+				for i, node := range graph.Nodes {
+					if node.Version.String() == edge.To {
+						target = &graph.Nodes[i]
+						break
+					}
+				}
+				if target == nil {
+					return current, updates, nil, &Error{
+						Reason:  "ResponseInvalid",
+						Message: fmt.Sprintf("no node for conditional update %s", edge.To),
+					}
+				}
+				update, err := convertRetrievedUpdateToRelease(*target)
+				if err != nil {
+					return current, updates, nil, &Error{
+						Reason:  "ResponseInvalid",
+						Message: fmt.Sprintf("invalid conditional update node: %s", err),
+					}
+				}
+				conditionalUpdates = append(conditionalUpdates, configv1.ConditionalUpdate{
+					Release: update,
+					Risks:   conditionalEdges.Risks,
+				})
+			}
+		}
+	}
+
+	for i := len(updates) - 1; i >= 0; i-- {
+		for _, conditionalUpdate := range conditionalUpdates {
+			if conditionalUpdate.Release.Image == updates[i].Image {
+				klog.Warningf("Update to %s listed as both a conditional and unconditional update; preferring the conditional update.", conditionalUpdate.Release.Version)
+				updates = append(updates[:i], updates[i+1:]...)
+			}
+		}
+	}
+
+	if len(updates) == 0 {
+		updates = nil
+	}
+
+	for i := len(conditionalUpdates) - 1; i >= 0; i-- {
+		for j, risk := range conditionalUpdates[i].Risks {
+			conditionalUpdates[i].Risks[j].MatchingRules, err = clusterconditions.PruneInvalid(ctx, risk.MatchingRules)
+			if len(conditionalUpdates[i].Risks[j].MatchingRules) == 0 {
+				klog.Warningf("Conditional update to %s, risk %q, has empty pruned matchingRules; dropping this target to avoid upsetting the Kubernetes API server.  Pruning results: %s", conditionalUpdates[i].Release.Version, risk.Name, err)
```

Comment:
> upsetting

Seems a better description is needed. Results in an error or whatever.

Reply:
`upsetting` -> `rejections when pushing to` in b12e5ba9 -> c3a213d5